### PR TITLE
[Flang][OpenMP][Lower] Use clause operand structures

### PR DESCRIPTION
**Pull request by skatrak**
This patch updates Flang lowering to use the new set of OpenMP clause operand structures and their groupings into directive-specific sets of clause operands.

It simplifies the passing of information from the clause processor and the creation of operations.

The `DataSharingProcessor` is slightly modified to not hold delayed privatization state. Instead, optional arguments are added to `processStep1` which are only passed when delayed privatization is used. This enables using the clause operand structure for `private` and removes the need for the ad-hoc `DelayedPrivatizationInfo` structure.

The processing of the `schedule` clause is updated to process the `chunk` modifier rather than requiring two separate calls to the `ClauseProcessor`.

Lowering of a block-associated `ordered` construct is updated to emit a TODO error if the `simd` clause is specified, since it is not currently supported by the `ClauseProcessor` or later compilation stages.

Removed processing of `schedule` from `omp.simdloop`, as it doesn't apply to `simd` constructs.

### CODE REVIEW

**Review by kparzysz — APPROVED** [review]
LGTM

**Comment by kparzysz** [comment]
Can this be merged?

**Comment by skatrak** [comment]
> Can this be merged?

I waited a few days in case there was more feedback, but at the moment I'm just waiting for #88206 to land before I update it and merge it.

**Pull request merged**
